services/sqs/src/main/java/software/amazon/awssdk/services/sqs/internal/batchmanager/ReceiveMessageBatchManager.java

```diff
@@ -39,19 +38,21 @@ public class ReceiveMessageBatchManager implements SdkAutoCloseable {
 
     public ReceiveMessageBatchManager(SqsAsyncClient sqsClient,
                                       ScheduledExecutorService executor,
-                                      BatchOverrideConfiguration config) {
+                                      ResponseBatchConfiguration config) {
         this.sqsClient = sqsClient;
         this.executor = executor;
-        this.config = new ResponseBatchConfiguration(config);
+        this.config = config;
+
+
     }
 
     public CompletableFuture<ReceiveMessageResponse> batchRequest(ReceiveMessageRequest request) {
-        if (canBeRetrievedFromQueueBuffer(request)) {
+        String ineligibleReason = checkBatchingEligibility(request);
+        if (ineligibleReason == null) {
             return receiveBatchManagerMap.computeIfAbsent(generateBatchKey(request), key -> createReceiveBatchManager(request))
                                          .processRequest(request);
         } else {
-            log.debug(() -> "canBeRetrievedFromQueueBuffer failed, so skipping batching for request for Queue with URL: "
-                            + request.queueUrl());
+            log.debug(() -> String.format("Batching skipped. Reason: %s", ineligibleReason));
```

Comment:
nit: maybe call checkBatchingEligibilty here, i.e,`log.debug(() -> String.format("Batching skipped. Reason: %s",   checkBatchingEligibility(request)))` so that it only gets called if debug level is enabled.